include/boost/geometry/strategies/cartesian/buffer_join_miter.hpp

```diff
@@ -89,8 +89,8 @@ class join_miter
         // Check the distance ip-vertex (= miter distance)
         // (We calculate it manually (not using Pythagoras strategy) to reuse
         //  dx and dy)
-        coordinate_type const dx = get<0>(p) - get<0>(vertex);
-        coordinate_type const dy = get<1>(p) - get<1>(vertex);
+        promoted_type const dx = get<0>(p) - get<0>(vertex);
+        promoted_type const dy = get<1>(p) - get<1>(vertex);
```

Comment:
Even though the probability is a lot smaller this could still cause overflow. It's ok for now but we could think about `determinant()` function which would internally perform integral computation using bigger types, potentially some simplified form of multiprecision.

Reply:
Or you could first convert to `promoted_type` and subtract afterwards. With this the precision would be lesser but there wouldn't be an overflow in subtraction.

Reply:
Btw, there would be a warning here about implicit converson from `int` to `double`.

Reply:
Thanks for your comments. Yes, there could be an overflow in subtraction if coordinates are huge and both plus and minus. I'll create an issue but will not handle it immediately. We probably have many of these kind of places in the library.